src/resmom/stage_func.c

```diff
@@ -1961,33 +1995,40 @@ sys_copy(int dir, int rmtflg, char *owner, char *src, struct rqfpair *pair, int
 			if ((pw->pw_userlogin == INVALID_HANDLE_VALUE) && (strcmpi(owner, getlogin()) == 0)) {
 				sprintf(log_buffer, "CreateProcess(%s) under acct %s wdir=%s",
 					cmd_line, owner, wdir);
-				log_event(PBSEVENT_DEBUG, PBS_EVENTCLASS_JOB, LOG_DEBUG, "sys_copy", log_buffer);
+				log_event(PBSEVENT_DEBUG, PBS_EVENTCLASS_JOB, LOG_DEBUG, __func__, log_buffer);
 				rc = CreateProcess(NULL, cmd_line,
 					NULL, NULL, TRUE, flags,
 					NULL, wdir, &si, &pi);
+				if (rc == 0) {
+					errno = GetLastError();
+					log_err(-1, __func__, "CreateProcess failed");
+				}
 			} else {
 				sprintf(log_buffer, "CreateProcessAsUser(%d, %s) under acct %s wdir=%s",
 					pw->pw_userlogin, cmd_line, getlogin(), wdir);
-				log_event(PBSEVENT_DEBUG, PBS_EVENTCLASS_JOB, LOG_DEBUG, "sys_copy", log_buffer);
+				log_event(PBSEVENT_DEBUG, PBS_EVENTCLASS_JOB, LOG_DEBUG, __func__, log_buffer);
 				rc = CreateProcessAsUser(pw->pw_userlogin, NULL, cmd_line,
 					NULL, NULL, TRUE, flags,
 					NULL, wdir, &si, &pi);
+				if (rc == 0) {
+					errno = GetLastError();
+					log_err(-1, __func__, "CreateProcessAsUser failed");
+				}
 			}
 
-			if (rc == 0)
-				errno = GetLastError();
-
 			close(fd);
 			fd = -1; /* already done the close */
 
 			if (errno != 0) {
 				rc = errno + 10000;	/* error on fork (100xx), retry */
 			} else {
 				if (WaitForSingleObject(pi.hProcess,
-					INFINITE) != WAIT_OBJECT_0)
-					errno = GetLastError();
-				else if (!GetExitCodeProcess(pi.hProcess, &rc))
-					errno = GetLastError();
+					INFINITE) != WAIT_OBJECT_0) {
+					log_errf(-1, __func__, "WaitForSingleObject failed with status=%d", rc);
+				}
```

Comment:
https://docs.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-waitforsingleobject
we should call GetLastError only for WAIT_FAILED, for other error code please use log_eventf and log the actual code returned.
You can refer [this change](https://github.com/PBSPro/pbspro/blob/bf759bd248a0b7c55d50c76cfec11d45ced92261/src/lib/Libwin/popen.c#L223).